### DISCUSSION TITLE
How to save changes in .sqlite file or disable WAL-mode?

### DISCUSSION THREAD

**Discussion by ghcusr** [pawelsalawa/sqlitestudio#3853] [Q&A]
I need to modify the .sqlite file a little (just add some string in a table) When I do it all changes save in a new created .sqlite-wal file. How can I save added strings in original .sqlite file?

**Comment by pawelsalawa** [comment]
WAL file should exist only until you commit transaction. Did you manually start a transaction? (i.e. did you type `BEGIN` or 'SAVEPOINT` in your SQL queries?)

Normally all changes - once committed - are saved to the original file, as expected.

**Comment by gabecook** [comment]
Hi, I'm having the same issue.

Steps to reproduce.

1. Open a database.
2. Run a delete query. Something like
  `DELETE FROM foo;`
3. Verify in the data view that foo has no records.
4. Close the connection to the database.
5. Close SQLiteStudio.
6. There will be temp files left behind. In my case, db-shm and db-wal.
7. Delete these files.
8. The table will have the data again. The changes were not saved into the database.

This is in version v3.2.1

**Reply by gabecook** [reply]
OH! Also thank you for this really amazing tool! It is very helpful.

**Comment by gabecook** [comment]
There is an issue for this here: https://github.com/pawelsalawa/sqlitestudio/issues/3875

**Comment by maxima120** [comment]
I have same issue all the time on all projects. If I open a db in the SqliteStudio and make changes - I always have wal file even if I close the studio. I dont not use any transactions.

**Comment by pawelsalawa** [comment]
Similar problem in #4367. Solution:
`pragma journal_mode = delete`
